### PR TITLE
Milsestone 10: Add api

### PR DESCRIPTION
**Pull request by iKrystyan**
- [x] Create an API endpoint to list all posts for a user.
- [x] Create an API endpoint to list all comments for a user's post.
- [x] Create an API endpoint to add a comment to a post.
- [x] Add authentication to your API endpoints.
- [x] Your API endpoints should receive JSON and respond JSON as well.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -7,4 +7,16 @@
     end
   end
   root to: "users#index"
+
+  namespace :api do
+    namespace :v1 do
+      post :auth, to: "authentication#login"
+      resources :users do
+        post :create, to: "users#create"
```
Comment (Ismailco) [inline config/routes.rb]: - [ ] Your app should have an API endpoint for authentication, both registration, and login, kindly make sure to use the same endpoints as follow so your app is consistent with the requirements:
* Login endpoint - **POST :** `/login`
* Registration endpoint - **POST :** `/register` or `/signup`

File: config/routes.rb [hunk 2]
```diff
@@ -7,4 +7,16 @@
...
+        resources :posts do
+          resources :comments
```
Comment (Ismailco) [inline config/routes.rb]: - [ ] [OPTIONAL] It would be helpful to add the API ENDPOINTS to your README file so your API is well documented

**Review by Ismailco — CHANGES_REQUESTED** [review]
### Hi @Christianib003, 

Good job so far 👍🏼 
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✔️  No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @PRATAP-KUMAR, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**